crates/script/src/lib.rs

```diff
@@ -621,6 +623,37 @@ impl ScriptConfig {
             .legacy_assertions(self.config.legacy_assertions);
 
         let use_zk = self.config.zksync.run_in_zk_mode();
+        let mut maybe_zk_env = None;
+        if use_zk {
+            if let Some(fork_url) = &self.evm_opts.fork_url {
+                let provider =
+                    zksync_provider().with_recommended_fillers().on_http(fork_url.parse()?);
+                let maybe_details =
+                    provider.get_block_details(env.block.number.try_into()?).await?;
```

Comment:
I think we should be using [getFeeParams](https://docs.zksync.io/zksync-protocol/api/zks-rpc#zks_getfeeparams) here instead?

Reply:
Not sure, for two reasons: 

1. This tries to mimic what is done on evm which is using the values from the last block to get  the environment. See: https://github.com/matter-labs/foundry-zksync/blob/a4f0ebf8ea8f0725196b051b89dcf0ef7c49caed/crates/evm/core/src/fork/init.rs#L65
Also `getFeeParams` 
2. Both `era_test_node` and `anvil_zksync` are returning `{"jsonrpc":"2.0","error":{"code":-32601,"message":"Method not found"},"id":1}` for this rpc call. If this is the way to go maybe we want to address making the rpc call work for it before.

Maybe there's something I am missing though.

Reply:
I see, I confirmed it should be `getFeeParams` but it is not supported in `anvil-zksync` yet. We can proceed with using block details until supported is added there.